### ISSUE TITLE
Use Typescript Decorators for binding this to class Methods.

### ISSUE THREAD

**Issue by akgupta0777** [electron/fiddle#1027]
Currently in a lot of files, Class Methods are defined and binded with `this` keyword so that the correct object gets referenced on calling.
An example of this - 
https://github.com/electron/fiddle/blob/50b6656690801882a9fc907669de3002199e4d9b/src/renderer/bisect.ts#L9-L12

Since project is written in TypeScript we can use typescript **experimental decorators** on class Methods that will automatically bind `this` keyword to the functions attached with the decorator.

This is a great improvement in code quality and also we can reduce a lot of lines of code.

**Comment by punndcoder28** [comment]
Hey. I'd like to take this up if this has not already been solved or if you guys have still not moved into functional component. I propose we write a decorator function like this
```
const bind = (context) => {
  return (target, memberName, propertyDescriptor) => {
    get() {
      const boundedFunction = propertyDescriptor.value.bind(context)
      Object.defineProperty(context, memberName, {
        value: boundedFunction
      })
      return boundedFunction
    }
  }
}
```

Do let me know if this is the right way and would be happy to contribute

**Comment by Swagat47** [comment]
Hey I'd like to take up this issue is it has not already been solved.

`function bound(originalMethod: any, context: ClassMethodDecoratorContext) {
    const methodName = context.name;
    if (context.private) {
        throw new Error(`'bound' cannot decorate private properties like ${methodName as string}.`);
    }
    context.addInitializer(function () {
        this[methodName] = this[methodName].bind(this);
    });
}`

Let me know if its the right way to do this, I would be happy to contribute.

**Comment by prajwal-26** [comment]
import { RunnableVersion } from '../interfaces';
import { autobind } from 'bind-decorator';

@autobind
export class Bisector {
  public revList: Array<RunnableVersion>;
  public minRev: number;
  public maxRev: number;
  private pivot: number;

  constructor(revList: Array<RunnableVersion>) {
    this.revList = revList;
    this.minRev = 0;
    this.maxRev = revList.length - 1;
    this.calculatePivot();
  }


Changes Made:

Added the @autobind decorator from the bind-decorator library to automatically bind class methods.
Removed manual method binding using .bind(this) for improved code readability and maintainability.
Context:

This pull request addresses the issue https://github.com/electron/fiddle/issues/1027 which requested the Use Typescript Decorators for binding this to class Methods.

How to Test:

Clone this PR branch.
Install the autobind-decorator library by running the following command:
Additional Notes:

Please review the changes and provide feedback.
Let me know if any further adjustments or clarifications are needed.

**Comment by bobbysharma05** [comment]
Hey I think, I have understood the issue and wan to resolve it. Kindly Assign this issue to me.

**Comment by jayzalani** [comment]
```
import { autobind } from 'bind-decorator';

@autobind
export class Bisector {
  public revList: Array<RunnableVersion>;
  public minRev: number;
  public maxRev: number;
  private pivot: number;

  constructor(revList: Array<RunnableVersion>) {
    this.revList = revList;
    this.minRev = 0;
    this.maxRev = revList.length - 1;
    this.calculatePivot();
  }
```
I think this will be more right approach to solve the problem let me know if this is correct or not @codebytere and @dsanders11

**Comment by dsanders11** [comment]
I don't believe we actually want to pursue this any more, so I'm going to close the issue.

**Comment by jayzalani** [comment]
ohh that's great so I'm planning to get some contribution for GSOC can please tell me if there any issues that I can contribute @dsanders11
